src/deploy/campaign/environments.ts

```diff
@@ -117,8 +119,56 @@ export const getConfig = (
     rootPriceConfig: priceConfig,
     zeroVaultAddress: process.env.ZERO_VAULT_ADDRESS ? process.env.ZERO_VAULT_ADDRESS : zeroVault.address,
     mockMeowToken: process.env.MOCK_MEOW_TOKEN ? !!process.env.MOCK_MEOW_TOKEN : true,
-    stakingTokenAddress: process.env.STAKING_TOKEN_ADDRESS ? process.env.STAKING_TOKEN_ADDRESS : MEOW_TOKEN
+    stakingTokenAddress: process.env.STAKING_TOKEN_ADDRESS ? process.env.STAKING_TOKEN_ADDRESS : MEOW_TOKEN,
   };
 
+  // Will throw an error based on any invalid setup, given the `ENV_LEVEL` set
+  validate(config);
+
   return config;
 };
+
+// For testing the behaviour when we manipulate, we have an optional "env" string param
+export const validate = (config : IDeployCampaignConfig, env ?: string, mongoUri ?: string) => {
+  let envLevel;
+
+  if (env) {
+    envLevel = env;
+  } else {
+    envLevel = process.env.ENV_LEVEL;
```

Comment:
this assumes it will always be set, but this can be undefined as well. we should check that

Reply:
The below lines explicitly check for `dev`, `test`, and `prod`, but if none of those are satisfied (the case that you've specified) we call the revert message